docs/tutorials/conntrack.rst

```diff
@@ -0,0 +1,353 @@
+<https://ryu.readthedocs.io/en/latest/nicira_ext_ref.html?highlight=translation#ryu.ofproto.ofproto_v1_3_parser.NXActionNAT>`_.
+
+The `Firewalling Actions section
+<https://man7.org/linux/man-pages/man7/ovs-actions.7.html#FIREWALLING_ACTIONS>`_ of the
+`ovs-actions(7)` man page is a helpful reference to understand how the NAT action
+behaves. 
+
+Now we augment the ACLs from the previous example with an additional `nat` option,
+replacing them with the following:
+
+.. code-block:: yaml
+    :caption: /etc/faucet/faucet.yaml
+
+    acls:
+        conntrack_fw:
+            - rule:
+                eth_type: 0x0806 # arp
+                actions:
+                    allow: True
+            - rule:
+                eth_type: 0x0800 # ipv4
+                ct_state: 0/0x20  # match -trk (untracked)
+                actions:
+                    ct:
+                        zone: 10
+                        table: 0
+            - rule:
+                eth_type: 0x0800 # ipv4
+                ipv4_src: 10.0.0.1
+                ipv4_dst: 10.0.1.2
+                ct_state: 0x21/0x21 # match +new - packets to establish a new connection
+                actions:
+                    ct:
+                        zone: 10
+                        flags: 1  # "commit" the new connection
+                        table: 1
+                        # sNAT the connection to the faucet VIP
+                        nat:
+                            flags: 1
+                            range_ipv4_min: 10.0.0.254
+                            range_ipv4_max: 10.0.0.254
+            - rule:
+                eth_type: 0x0800 # ipv4
+                ct_zone: 10
+                ct_state: 0x22/0x22 # match +est - packets in an established connection
+                actions:
+                    ct:
+                        zone: 10
+                        flags: 1  # NAT must include "commit" - this is a NO-OP for existing connections
+                        table: 1
+                        # sNAT the packets in an existing connection appropriately according to their direction
+                        nat:
+                            flags: 1
+            - rule:
+                eth_type: 0x0800 # ipv4
+                ipv4_src: 10.0.1.2
+                ipv4_dst: 10.0.0.1
+                actions:
+                    allow: False
+
+Reload Faucet to apply the new configuration.
+
+.. code:: console
+
+    sudo systemctl reload faucet
+
+We can now see how OVS + conntrack will NAT the packets:
+
+.. code:: console
+
+    ovs-appctl ofproto/trace br0 in_port=1,tcp,nw_src=10.0.0.1,nw_dst=10.0.1.2
+
+Our ping from host1 to host2 should continue to work, establishing an entry in the
+connection tracker. This time, however, host1's source IP of 10.0.0.1 gets NATed to the
+faucet VIP of 10.0.0.254.
+
+.. code:: console
+
+    as_ns host1 ping 10.0.1.2
+
+    tcpdump -n -e -ttt -i veth-host2 10.0.0.254
+
+
+ICMPv6 and ARP filtering
```

Comment:
Remove this section and it can be added in a subsequent PR.

Reply:
That's a good idea for now. I was thinking you have a better understanding on what to express here anyways.